tests/unit/test__init__.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from unittest import mock
+
+import pytest
+
+# Module paths used for mocking
+MODULE_PATH = "db_dtypes"
+HELPER_MODULE_PATH = f"{MODULE_PATH}._versions_helpers"
+MOCK_EXTRACT_VERSION = f"{HELPER_MODULE_PATH}.extract_runtime_version"
+MOCK_WARN = "warnings.warn"  # Target the standard warnings module
+
+
+@pytest.mark.parametrize(
+    "mock_version_tuple, version_str",
+    [
+        ((3, 7, 10), "3.7.10"),
+        ((3, 7, 0), "3.7.0"),
+        ((3, 8, 5), "3.8.5"),
+        ((3, 8, 12), "3.8.12"),
+    ],
+)
+def test_check_python_version_warns_on_unsupported(mock_version_tuple, version_str):
+    """
+    Test that _check_python_version issues a FutureWarning for Python 3.7/3.8.
+    """
+
+    from db_dtypes import _check_python_version
+
+    # Mock the helper function it calls and the warnings.warn function
+    with mock.patch(MOCK_EXTRACT_VERSION, return_value=mock_version_tuple), mock.patch(
+        MOCK_WARN
+    ) as mock_warn_call:
```

Comment:
I'm curious if you've tried `pytest.warns` context manager rather than mocking out warnings.warn? https://docs.pytest.org/en/stable/how-to/capture-warnings.html#warns

Reply:
I have used the pytest.warns context manager in several places. During troubleshooting this, we tried many, many approaches and all of them failed so we swapped out a ton of things to try and find a solution that ended up working.

* dependencies and import timing will be the death of me.